src/history/hs_rec.c

```diff
@@ -686,7 +686,8 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
              * inserting the first update.
              */
             nentries = MAX_REVERSE_MODIFY_NUM;
-            if (upd->type == WT_UPDATE_MODIFY && enable_reverse_modify &&
+            if (!F_ISSET(upd, WT_UPDATE_DS) && upd->type == WT_UPDATE_MODIFY &&
```

Comment:
I believe a comment around why we alway want to insert a previous datastore value as a full update will be helpful for future reference.

Reply:
Sure will update the comment to explain the reason behind.

Reply:
Added a comment.